### PR TITLE
TOR: config folder location

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: net-vpn/tor/patches/tor-0.3.3.9.patchset [hunk 1]
```diff
@@ -35,5 +35,45 @@ index 9c88d37..74f2e48 100644
...
++++ b/src/or/config.c
+@@ -5024,7 +5024,11 @@ find_torrc_filename(config_line_t *cmd_arg,
+ #ifndef _WIN32
+       char *fn = NULL;
+       if (!defaults_file) {
+-        fn = expand_filename("~/.torrc");
++        #ifndef __HAIKU__
++          fn = expand_filename("~/.torrc");
++        #else
++          fn = expand_filename("~/config/settings/tor/torrc");
```
Comment (alaviss) [inline net-vpn/tor/patches/tor-0.3.3.9.patchset]: Please consider using `find_directory()` instead.

File: net-vpn/tor/patches/tor-0.3.3.9.patchset [hunk 2]
```diff
@@ -35,5 +35,45 @@ index 9c88d37..74f2e48 100644
...
+         file_status_t hmst = file_status(fn);
+@@ -7749,7 +7753,11 @@ get_data_directory(const char *val)
+ #else /* !(defined(_WIN32)) */
+   const char *d = val;
+   if (!d)
+-    d = "~/.tor";
++    #ifndef __HAIKU__
++      d = "~/.tor";
++    #else
++      d = "~/config/settings/tor";
```
Comment (alaviss) [inline net-vpn/tor/patches/tor-0.3.3.9.patchset]: IMO `B_USER_VAR_DIRECTORY` might be better suited for storing data from applications.

File: net-vpn/tor/patches/tor-0.3.3.9.patchset (hunk 1 above)
Comment (scottmc) [reply net-vpn/tor/patches/tor-0.3.3.9.patchset]: I second that request.  Also I'd change the logic here to ```#ifdef __HAIKU__``` and if not let it fall thru to the else for the defaul case.  This makes it easier for other OSes that might also need to move the settings file, to then add a case for their OS as well.

File: net-vpn/tor/patches/tor-0.3.3.9.patchset (hunk 2 above)
Comment (scottmc) [reply net-vpn/tor/patches/tor-0.3.3.9.patchset]: same logic comment as about...  change to ```ifdef __HAIKU__``` and if not let the else be the default.

**Comment by scottmc** [comment]
@fbrosson i see you bumped the version a couple days ago.  can you review this open pull request to see if anything in here should have been included in your update?

File: net-vpn/tor/patches/tor-0.3.3.9.patchset (hunk 1 above)
Comment (fbrosson) [reply net-vpn/tor/patches/tor-0.3.3.9.patchset]: Same comments as @alaviss and @scottmc :)

**Comment by extrowerk** [comment]
I never had too much luck using find_directoy(), that's why i never using it.

**Comment by scottmc** [comment]
Ok, then let's close this PR, and instead open a new issue to solve this directory issue using find_directory().  This might be a good GCI task.

**Pull request closed without merging**